src/seq.rs

```diff
@@ -141,16 +145,17 @@ impl Seq {
     /// Note: this function is a non-allocating version of event_input.
     ///
     /// In case of sysex or other custom data length message, the supplied callback will be called with the buffer.
-    /// The return value of the callback is only relevant if you intend to call get_data::<Vec<u8>>() on the returned Event.
-    pub fn event_input_cb<F: FnOnce(EventType, &[u8]) -> Option<Vec<u8>>>(&self, f: F) -> Result<Event> {
+    /// The return value of the callback is only relevant if you intend to call get_data::<Vec<u8>>() on the returned Event
+    /// or the event is passed back unchanged to ALSA for output.
+    pub fn event_input_cb<'a, 'b, F: FnOnce(EventType, &[u8]) -> Option<Vec<u8>>>(&'a self, f: F) -> Result<Event<'b>> {
         let mut z = ptr::null_mut();
         try!(acheck!(snd_seq_event_input(self.0, &mut z)));
         unsafe { Event::extract(&mut *z, "snd_seq_event_input", f) }
     }
 
     /// Note: this function will allocate in case of sysex or other custom data length messages.
-    /// If this is undesired, you can use event_input_cb instead.
-    pub fn event_input(&self) -> Result<Event> {
+    /// If this is undesired, you can use `event_input_cb` instead.
+    pub fn event_input<'a, 'b>(&'a self) -> Result<Event<'b>> {
```

Comment:
You're creating a lifetime out of thin air here. You really mean `Event<'static>`, right?